src/nvim/charset.c

```diff
@@ -231,7 +231,7 @@ int buf_init_chartab(buf_T *buf, int global)
           if (i == 0) {
             // (re)set ID flag
             if (tilde) {
-              chartab[c] &= ~CT_ID_CHAR;
+              chartab[c] &= (char_u)~CT_ID_CHAR;
```

Comment:
as `chartab[c]` is a 8bit bitset I think the proper type should be `uint8_t` (change the casts in this function as well as the declaration in `globals.h`)